### PR TITLE
♻️ 🔧 Refactor app context

### PR DESCRIPTION
**Pull request by anncatton**
Separate secret retrieval from env AppConfig to remove the need for async `getAppConfig` calls. Adding these changes to help with the refactoring for #353 
- creates secrets context file
- removes config arg from State class init
- use config and secrets args in Storage class init
- removes config arg from calculation utils, service calls
- fixes tests. Adds one mock config for all state and utils tests, there were inconsistent configs being used
- removes some renewal tests that won't be relevant with new renewal flow
- reorganizes service file into folder structure for different service types

### CODE REVIEW

File: src/secrets.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  mongoProperties: MongoSecrets;
+}
+
+dotenv.config();
+let secrets: AppSecrets | undefined = undefined;
+
+const vaultEnabled = (process.env.VAULT_ENABLED || '').toLowerCase() === 'true';
+
+const loadVaultSecrets = async () => {
+  console.info('Loading Vault secrets...');
```
Comment (joneubank) [inline src/secrets.ts:34]: use logger

File: src/secrets.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  try {
+    if (process.env.VAULT_SECRETS_PATH) {
+      return await vault.loadSecret(process.env.VAULT_SECRETS_PATH);
+    }
+    throw new Error('Path to secrets not specified but vault is enabled');
+  } catch (err) {
+    console.error(err);
```
Comment (joneubank) [inline src/secrets.ts:41]: logger

**Review by joneubank — APPROVED** [review]
This is a big improvement, thanks.

Can we swap the few uses of `console.log` for the logger? I think you copy/pasted these from the original place, but still...

File: src/secrets.ts (hunk 2 above)
Comment (anncatton) [reply src/secrets.ts:41]: i sure did copy paste. will fix

**Pull request merged**